### PR TITLE
Updates got-couch to 1.1.2

### PR DESCRIPTION
**Pull request by mszymulanski**

### CODE REVIEW

**Review by jxbadam — APPROVED** [review]
:shipit: :fire:

File: lib/couchdb/index.js [hunk 1]
```diff
@@ -119,6 +120,9 @@ const findById = R.curry((couch, bucket, projection, id) => R.composeP(
 )(bucket, id))
 
 
+const del = R.curry((couch, bucket, id, rev) => couch.del(bucket, id, rev))
```
Comment (jonajonlee) [inline lib/couchdb/index.js]: this should be deleteById

Comment (jonajonlee) [reply lib/couchdb/index.js]: probably something along the lines of this

```javascript
const deleteById = R.curry((couch, bucket, id) =>
  couch.getById(couch, id)
  .then(R.prop('_rev_id')
  .then(couch.deleteByIdAndRevId(couch, bucket, id))
)

File: lib/couchdb/wrapper.js [hunk 2]
```diff
@@ -50,10 +50,16 @@ const list = wrapper((couch, bucket, params) =>
 )
 
 
+const del = wrapper((couch, bucket, id, rev) =>
```
Comment (jonajonlee) [inline lib/couchdb/wrapper.js]: this should be called deleteByIdAndRevId

File: lib/memory/index.js [hunk 3]
```diff
@@ -173,11 +173,22 @@ module.exports = function MemoryStore(data_set) {
   )
 
 
+  const del = R.curry((bucket, id) =>
```
Comment (jonajonlee) [inline lib/memory/index.js]: Change all of these `del` functions of the Kuss interfaces to `deleteById` to be more explicit

File: lib/couchdb/index.js (hunk 1 above)
Comment (jonajonlee) [reply lib/couchdb/index.js]: it should also throw the NotFound error if getById returns nothing.

File: lib/memory/index.js [hunk 4]
```diff
@@ -173,11 +173,22 @@ module.exports = function MemoryStore(data_set) {
...
+      getById(bucket, id)
+      .tap(R.when(R.isNil, () => Err.NotFound.throw(STORE, bucket, id)))
```
Comment (jonajonlee) [inline lib/memory/index.js:185]: 👍

File: lib/couchdb/index.js (hunk 1 above)
Comment (jonajonlee) [reply lib/couchdb/index.js]: noticed the MemoryStore.deleteById returns true on successful delete. This one should also return true in the same case

**Review by jonajonlee — CHANGES_REQUESTED** [review]

**Pull request merged**